docs/dalloc.md

```diff
@@ -129,33 +133,33 @@ descriptor which has a bitmap of which pages are allocated within it or not.
 
 The id of the CPU core on which the thread is currently running decides what
 arena to use. This ensures a low level of contention in practice, and will limit
-the number of arena used to twice the number of core on the machine.
+the number of arenas used to twice the number of cores on the machine.
 
 ### Large allocations
 
-All the block descriptors which have any slots left are kept in the serie of
+All the block descriptors which have any slots left are kept in the series of
 pairing heaps, based on the maximum size class that can be allocated. The
 pairing heaps allows for amortized constant time insertions and deletions. This
-way, finding a block to allocate from can be extremely quick.
+way, finding a block from which to allocate can be extremely quick.
 
 ### Small allocations
 
 For each small size class, the arena contains a bin from which small allocations
-can be done. Each bin contains a pairing heap of all the non full slab of that
-size class that the arena handles. In case a bin ran out of slabs, a new slab is
+can be done. Each bin contains a pairing heap of all the non-full Slabs of that
+size class that the arena handles. In case a bin runs out of Slabs, a new Slab is
 allocated using the same strategy as large allocations.
 
 ## Thread Cache
 
-**/!\ Thread cache are not fully implemented as this time!**
+**/!\ Thread caches are not fully implemented as this time!**
 
-Each thread that allocates memory has a thread local cache. Small allocations
+Each thread that allocates memory has a *Thread Local Cache*. Small allocations
 are typically served from the cache, and the cache is refilled in bulk from the
 arena when it runs out. When freeing elements, they are not actually freed, but
-instead put in the thread cache for later resue. Element are released in batch
+instead put in the thread cache for later reuse. Elements are released in batch
 when the cache grow too large.
 
 This mechnism ensures that most of allocations avoid any possible contention and
-are served as fast as possible. In addition, this improve the locality of
+are served as fast as possible. In addition, this improves the locality of
 allocation on a per thread basis as they come from the same batch of
-allocations, which reduce pressure ont he TLB.
+allocations, which reduces pressure on the TLB.
```

Comment:
Unsure what TLB means, so I left it.

Reply:
TLB is translation look-aside buffer. It is the part of the MMU that translates virtual addresses into physical.